src/_pytest/config/__init__.py

```diff
@@ -1078,12 +1085,21 @@ def _checkversion(self):
                     )
                 )
 
-    def _validatekeys(self):
+    def _validate_keys(self) -> None:
         for key in sorted(self._get_unknown_ini_keys()):
-            message = "Unknown config ini key: {}\n".format(key)
-            if self.known_args_namespace.strict_config:
-                fail(message, pytrace=False)
-            sys.stderr.write("WARNING: {}".format(message))
+            self._emit_warning_or_fail("Unknown config ini key: {}\n".format(key))
+
+    def _validate_plugins(self) -> None:
+        for plugin in self.getini("require_plugins"):
+            if not self.pluginmanager.hasplugin(plugin):
+                self._emit_warning_or_fail(
+                    "Missing required plugin: {}\n".format(plugin)
```

Comment:
@nicoddemus - I feel like it's unlikely, but a suite could still potentially work without a required plugn. For example, you could have required [pytest-greendots](https://pypi.org/project/pytest-greendots/), but since it has no configuration options that are required the suite will work with and without the plugin

Reply:
Indeed, but I think it is reasonable to stop executing if a plugin listed in `required_plugins` is not really installed.

Reply:
Now that I think about it again, I agree with you @nicoddemus. I mean, if it emits a warning but is 'required' I think that's pretty confusing. Let me change it to always stop executing. Any objections @RonnyPfannschmidt

Reply:
Great!

Make sure to display all missing plugins, instead of stopping on the first one missing.